tests/mapper/test_relational_node_mappers.py

```diff
@@ -0,0 +1,191 @@
+from stellargraph.core.graph import *
+from stellargraph.mapper.node_mappers import RelationalFullBatchNodeGenerator
+
+import networkx as nx
+import numpy as np
+import pytest
+import pandas as pd
+import scipy.sparse as sps
+
+
+def create_graph_features():
+    G = nx.MultiDiGraph()
+    G.add_nodes_from(["a", "b", "c"])
+    G.add_edges_from([("a", "b", "r1"), ("b", "c", "r1"), ("a", "c", "r2")])
+    return G, np.array([[1, 1], [1, 0], [0, 1]])
+
+
+class Test_RelationalFullBatchNodeGenerator:
+    """
+    Tests of FullBatchNodeGenerator class
+    """
+
+    n_feat = 2
+    target_dim = 5
+
+    gnx, features = create_graph_features()
+    nodes = list(gnx.nodes)
+    node_features = pd.DataFrame.from_dict(
+        {n: f for n, f in zip(nodes, features)}, orient="index"
+    )
+    G = StellarDiGraph(gnx, node_features=node_features)
+    N = len(G.nodes())
+    edge_types = sorted(set(e[-1] for e in G.edges))
+    num_relationships = len(edge_types)
+
+    def test_generator_constructor(self):
+
+        generator = RelationalFullBatchNodeGenerator(self.G)
+        assert len(generator.As) == self.num_relationships
+        assert all((A.shape == (self.N, self.N)) for A in generator.As)
+
+        assert generator.features.shape == (self.N, self.n_feat)
+
+    def test_generator_constructor_wrong_G_type(self):
+        with pytest.raises(TypeError):
+            generator = RelationalFullBatchNodeGenerator(nx.Graph(self.G))
```

Comment:
Can we add tests for other graph types? Should nx.DiGraph or nx.MultidiGraph (the type of nx graph you start with in `create_graph_features` work?

Reply:
This test is for making sure that passing any networkx graph to the generator will fail, only `StellarGraphBase` objects should be accepted.